common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -46,6 +46,11 @@
 
     private volatile Object result;
     private final EventExecutor executor;
+
+    // It is fine to not make this volatile as even if we override the value in there it does not matter as
+    // DefaultFutureCompletionStage has no state itself and is just a wrapper around this DefaultPromise instance.
+    private DefaultFutureCompletionStage<V> stage;
```

Comment:
Possible to use `FutureCompletionStage` instead of `DefaultFutureCompletionStage`?

Reply:
I did make it DefaultFutureCompletionStage to make it clear that it is fine to not use volatile (as it is just a delegate).